ARCs/arc-0027.md

```diff
@@ -0,0 +1,1611 @@
+  "type": "object",
+  "properties": {
+    "code": {
+      "type": "number",
+      "description": "An integer that defines the type of error"
+    },
+    "data": {
+      "type": "object",
+      "description": "Additional information about the error"
+    },
+    "message": {
+      "type": "string",
+      "description": "A human-readable message about the error"
+    },
+    "providerId": {
+      "type": "number",
+      "description": "A unique identifier for the provider"
+    }
+  },
+  "required": [
+    "code",
+    "message"
+  ]
+}
+```
+where:
+* `code`:
+  - **MUST** be a code of one of the [errors](#errors)
+* `message`:
+  - **SHOULD** be human-readable to allow for display to a user
+* `providerId`:
+  - **MUST** be a <a href="https://www.rfc-editor.org/rfc/rfc4122">UUIDv4</a> compliant string
+  - **MUST** be present if the error originates from the provider
+
+### Supported Methods
+
+#### Get Providers
+
+Sent by a dapp to get responses of any available providers. If the `params.providerId` property is supplied, only the provider with the matching ID **SHOULD** respond.
+
+This method is usually called before other methods as it allows the dapp to get the identity of providers, the networks the provider supports and the methods the provider supports on each network.
+
+For example, see [here](#get-providers-example).
+
+**Request Payload**
+
+```json
+{
+  "title": "Get Providers Request Payload",
+  "type": "object",
+  "properties": {
+    "id": {
+      "type": "string",
+      "description": "A globally unique identifier for the message"
+    },
+    "params": {
+      "type": "object",
+      "properties": {
+        "providerId": {
+          "type": "string",
+          "description": "A unique identifier for the provider"
+        }
+      }
+    },
+    "reference": {
+      "type": "string",
+      "description": "Identifies the purpose of the message"
+    }
+  },
+  "required": ["id", "reference"]
+}
+```
+where:
+* `params`:
+  - **OPTIONAL** if omitted, all providers **MAY** respond
+* `params.providerId`:
+  - **MUST** be a <a href="https://www.rfc-editor.org/rfc/rfc4122">UUIDv4</a> compliant string
+* `reference`:
+  - **MUST** be `arc0027:get_providers:request`
+
+**Response Payload**
+
+```json
+{
+  "title": "Get Providers Response Payload",
+  "type": "object",
+  "properties": {
+    "id": {
+      "type": "string",
+      "description": "A globally unique identifier for the message"
+    },
+    "reference": {
+      "type": "string",
+      "description": "Identifies the purpose of the message"
+    },
+    "requestId": {
+      "type": "string",
+      "description": "The ID of the request message"
+    },
+    "result": {
+      "type": "object",
+      "properties": {
+        "host": {
+          "type": "string",
+          "description": "A domain name of the provider"
+        },
+        "icon": {
+          "type": "string",
+          "description": "A URI pointing to an image"
+        },
```

Comment:
Can we force this to not be a remote URI? Ideally we could make it match https://github.com/wallet-standard/wallet-standard/blob/master/packages/core/base/src/wallet.ts#L24